crates/bitcoin-da/src/service.rs

```diff
@@ -588,6 +594,114 @@ impl DaService for BitcoinService {
+                            aggregate_idxs.push((tx_id, aggregate));
+                        }
+                    }
+                    ParsedLightClientTransaction::Chunk(_chunk) => {
+                        // we ignore them for now
+                    }
+                }
+            }
+        }
+
+        // collect aggregated txs from chunks
+        let mut aggregates = Vec::new();
+        for (tx_id, aggregate) in &aggregate_idxs {
+            let mut body = Vec::new();
+            let Ok(chunk_ids) = aggregate.txids() else {
+                anyhow::bail!("{}: Failed to get txids from aggregate", tx_id);
+            };
+            if chunk_ids.is_empty() {
+                anyhow::bail!("{}: Empty aggregate tx list", tx_id);
+            }
+            for chunk_id in chunk_ids {
+                let tx_res = {
+                    let exponential_backoff = ExponentialBackoff::default();
+                    retry_backoff(exponential_backoff, || async move {
+                        self.client
+                            .get_transaction(&chunk_id, None)
+                            .await
+                            .map_err(|e| {
+                                use bitcoincore_rpc::Error;
+                                match e {
+                                    Error::Io(_) => backoff::Error::transient(e),
+                                    _ => backoff::Error::permanent(e),
+                                }
+                            })
+                    })
+                    .await?
+                };
+                let tx = tx_res.transaction().map_err(|e| {
+                    anyhow::anyhow!(
+                        "{}:{}: Failed to decode the body of bitcoin tx {e}",
+                        tx_id,
+                        chunk_id
+                    )
+                })?;
+                let wrapped: TransactionWrapper = tx.into();
+                let parsed = parse_light_client_transaction(&wrapped).map_err(|e| {
+                    anyhow::anyhow!("{}:{}: Failed parse chunk: {e}", tx_id, chunk_id)
+                })?;
+                match parsed {
+                    ParsedLightClientTransaction::Chunk(part) => {
+                        body.extend(part.body);
+                    }
+                    ParsedLightClientTransaction::Complete(_)
+                    | ParsedLightClientTransaction::Aggregate(_) => {
+                        anyhow::bail!("{}:{}: Expected chunk, got other tx kind", tx_id, chunk_id);
+                    }
+                }
+            }
+            aggregates.push((*tx_id, body));
+        }
+
+        let mut result = Vec::new();
+        for (tx_id, blob) in completes.into_iter().chain(aggregates) {
+            let body = decompress_blob(&blob);
+            let data = DaDataLightClient::try_from_slice(&body)
+                .map_err(|e| anyhow::anyhow!("{}: Failed to parse body: {e}", tx_id))?;
+            result.push(data);
+        }
```

Comment:
we'd like to keep the ordering on these proofs

so if we have 

complete aggregate aggreagete complete on da

we want the node to process them in the same order